src/main/utility/utility.c

```diff
@@ -362,3 +362,19 @@ int return_code_for_signal(int signal) {
     // follow the behaviour of bash and add 128 to to the signal.
     return signal + 128;
 }
+
+void die_after_vfork() {
+    // Capture errno in a local, so that it can be examined in a stack trace.
+    int saved_errno = errno;
+
+    // Ensure our saved errno doesn't get optimized away.
+    asm volatile(/*asm=*/"" : /*outputs=*/ : /*inputs=*/"irm"(saved_errno));
+
+    // abort and kill are higher-level functions that could attempt to access
+    // global memory, which could have surprising results. We resort to a bare
+    // `kill`.
+    kill(getpid(), SIGABRT);
+
+    // Convince the compiler that we really don't return.
+    abort();
```

Comment:
What do you mean by "[...] and kill are higher-level functions that could attempt to access global memory [...] We resort to a bare `kill`"?

Do we ever expect the `kill()` to return and continue (I guess maybe if we have signal handlers)? Should the `abort()` be just `_exit()` instead?

Reply:
> What do you mean by "[...] and kill are higher-level functions that could attempt to access global memory [...] We resort to a bare kill"?

Oops - s/kill/raise/. Fixed.

> Do we ever expect the kill() to return and continue (I guess maybe if we have signal handlers)? Should the abort() be just _exit() instead to avoid any side-effects of abort() in that case?

No, we don't expect it. `kill` shouldn't return before the signal is delivered, so should never return. Went ahead and changed to _exit to be safe, though.